src/app/components/login/login.component.html

```diff
@@ -0,0 +1,15 @@
+<div
+  *ngIf="auth.user$ | async as user; else noUser"
+  style="vertical-align: middle;"
+>
+  <img [src]="user.photoURL" [alt]="user.displayName" class="avatar" />
+
+  <button mat-button color="accent" (click)="auth.logOut()">
+    SALIR
+  </button>
+</div>
+<ng-template #noUser>
+  <button mat-raised-button color="accent" (click)="auth.googleLogin()">
```

Comment:
If `googleLogin` is an async, network action, how do we handle errors here?

I propose something like this:
```html
<button (click)="googleLogin()">
```

```typescript
googleLogin() {
  this.authService.login().then(
    () => // login success, noop or redirect to /dashboard or something,
    err => // show error, problems at login try again later.
}
```

Reply:
What do you think about handling this promise on the service? Like this:

```
googleLogin() {
    this.afAuth
      .signInWithPopup(new auth.GoogleAuthProvider())
      .then((credential) => {
        this.snackbar.open(`🎉 You are logged in`, 'OK', matSnackbarOptions);
        this.updateUser(credential.user);
      })
      .catch((error) =>
        this.snackbar.open(
          `Something went wrong: ${error}`,
          'OK',
          matSnackbarOptions
        )
      );
  }
```

I think is better than having the promise on every component we need to login/logout but I'm not sure if I'm missing something. WDYT?

Reply:
Yeah, good thinking, much better!